gateway-eth-ts/src/service/GatewayNetwork.ts

```diff
@@ -0,0 +1,164 @@
+import { Provider } from "@ethersproject/providers";
+import {
+  GatewayNetwork,
+  GatewayNetwork__factory,
+  IGatewayGatekeeper,
+  IGatewayNetwork
+} from "../contracts/typechain-types";
+import { BigNumberish, ContractTransaction, Wallet, ethers } from "ethers";
+
+export enum GatewayNetworkFeatures {
```

Comment:
I am sure you know, but enums are discouraged from being used in typescript, it was added in early on and the typescript authors say if they could they would completely remove them

```
export const GATEWAY_NETWORK_FEATURES {
   REMOVE_GATEKEEPER_INVALIDATES_TOKENS
} as const

type ObjectVlaues<T> = T[keyof T]

type GatewayNetworkFeatures = ObjectValues<typeof GATEWAY_NETWORK_FEATURES>
```